docs/tasks/administer-cluster/configure-upgrade-etcd.md

```diff
@@ -16,47 +16,196 @@ redirect_from:
 
-**Warning**: Deviations from the assumptions are untested by continuous
-integration, and deviations might create undesirable consequences. Additional
-information about operating an etcd cluster is available [from the etcd
-maintainers](https://github.com/coreos/etcd/tree/master/Documentation).
+## Starting Kubernetes API server
+
+This section covers instruction on starting a Kubernetes API server with an etcd cluster in the deployment.
+
+### Single-node etcd cluster
+
+Use a single-node etcd cluster only for testing purpose.
+
+1. Run the following:
+
+  ``` bash
+   ./etcd --client-listen-urls=http://$PRIVATE_IP:2379 --client-advertise-urls=http://$PRIVATE_IP:2379
+  ```
+2. Start Kubernetes API server with the flag `--etcd-servers=$PRIVATE_IP:2379`.
+
+    Replace `PRIVATE_IP` with your etcd client IP.
+
+### Multi-node etcd cluster
+
+For durability and high availability, run etcd as a multi-node cluster in production and take its backup periodically. A five-member cluster is recommended in production. For more information, see [FAQ Documentation ](https://github.com/coreos/etcd/blob/master/Documentation/faq.md#what-is-failure-tolerance).
+
+Configure an etcd cluster either by static member information or by dynamic discovery. For more information on clustering, see [etcd Clustering Documentation](https://github.com/coreos/etcd/blob/master/Documentation/op-guide/clustering.md).
+
+For an example, consider a five-member etcd cluster running with the following client URLs: `http://$IP1:2379`, `http://$IP2:2379`, `http://$IP3:2379`, `http://$IP4:2379`, and `http://$IP5:2379`. To start a Kubernetes API server:
+
+1. Run the following:
+
+  ``` bash
+   ./etcd --client-listen-urls=http://$IP1:2379, http://$IP2:2379, http://$IP3:2379, http://$IP4:2379, http://$IP5:2379 --client-advertise-urls=http://$IP1:2379, http://$IP2:2379, http://$IP3:2379, http://$IP4:2379, http://$IP5:2379
+  ```
+
+2. Start Kubernetes API servers with the flag `--etcd-servers=$IP1:2379, $IP2:2379, $IP3:2379, $IP4:2379, $IP5:2379`.
+
+    Replace `IP` with your client IP addresses.
+
+###  Multi-node etcd cluster with load balancer
+
+To run a load balancing etcd cluster:
+
+1. Set up an etcd cluster.
+2. Configure a load balancer in front of the etcd cluster.
+   For example, let the address of the load balancer be `$LB`.
+3. Start Kubernetes API Servers with the flag `--etcd-servers=$LB:2379`.
+
+## Securing etcd clusters
+
+Access to etcd is equivalent to root permission in the cluster and ideally only the API server will have access to it. Considering the sensitivity of the data, it is recommended to grant permission to only those nodes that require access to etcd clusters.
```

Comment:
suggested edit: "Access to etcd is equivalent to root permission in the cluster, so ideally only the API server should have access to it."